### PR TITLE
CPRI-3964 | Create a Service for Taxonomies

### PR DESCRIPTION
**Pull request by rishab-mehra**
Hi Team,

I have added one service for placement taxonomies and will be adding some more services for other taxonomies

### CODE REVIEW

**Comment by propattern** [comment]
@rishab-mehra 
I don't think we should call these getXService.  The name of the service is already in the package, so not point appending the word Service to it.  just call it getXTaxonomy or getYTaxonomy.

Also these taxonomy fetch (get) files needs to be inside the taxonomy folders.

services/taxonomy/getXTaxonomy
services/taxonomy/getYTaxonomy

File: package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "husky": "0.14.3",
     "isomorphic-fetch": "^2.2.1",
     "prop-types": "15.6.1",
+    "proxyquire": "^2.1.0",
```
Comment (Kerenhb) [inline package.json]: I'm fine with proxyquire but did we discuss adding new packages?
@devu @propattern

File: src/services/__test__/getPlacementTaxonomyService.test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    id: 'https://schema.pearson.com/ns/placement/activity',
+    value: 'Activity'
+  },
+  {
+    id: 'https://schema.pearson.com/ns/placement/chapter',
+    value: 'Chapter'
+  },
+  {
+    id: 'https://schema.pearson.com/ns/placement/exercise',
+    value: '"Exercise"'
```
Comment (Kerenhb) [inline src/services/__test__/getPlacementTaxonomyService.test.js]: Why double quotes: '"

**Comment by rishab-mehra** [comment]
Right @propattern Actually it was in my mind as you have already told me. I missed it over.
 Thanks @propattern. I will add that change

File: src/services/__test__/getPlacementTaxonomyService.test.js (hunk 2 above)
Comment (rishab-mehra) [reply src/services/__test__/getPlacementTaxonomyService.test.js]: oops. single quotes will modify it @Kerenhb :)

File: package.json (hunk 1 above)
Comment (rishab-mehra) [reply package.json]: yes @Kerenhb As discussed with you I have used the rewire for mocking/stubbing the dependency, but today i was facing some issue in adding rewire plugin in .babelrc file and also in running some of its test conditions. So i am looking into parallelly also. Will add the test cases with rewire

Comment (devu) [reply package.json]: proxyquire is something we have used for mocking services in BP so,  we can go ahead with this one.

**Comment by devu** [comment]
If that's OK, I will add a startup sequence to provide a place to invoke those services as part of this PR. 
But this will need more works I believe, so far this taxonomy breaks.

File: package.json (hunk 1 above)
Comment (propattern) [reply package.json]: if this is the first time we are bringing a wiring library into this repo, then I recommend we use rewire which has a much neater interface compare to proxywire.
Much cleaner code and very simple.  It is used heavily in the components codebase.

Comment (devu) [reply package.json]: We tried rewire and things went wrong as they are not compatible with mocha. we use sinon, but proxywire was specifically introduced for services that we know was compatible. I said that before, I'm Ok with giving it a go but if someone will make it work witch mocha. We have no luxury of time for this, so we go for we know works.

Comment (propattern) [reply package.json]: I will give it a try in some future module soon.
Sinon is fine for creating the spies & stubs and they can still be used with rewire.

**Pull request merged**